frontend/src/pages/exploreApplication/ExploreApplications.tsx

```diff
@@ -17,6 +18,57 @@ import { useHistory } from 'react-router';
 
 const description = `Add optional applications to your Red Hat OpenShift Data Science instance.`;
 
+type ExploreApplicationsInnerProps = {
+  loaded: boolean;
+  isEmpty: boolean;
+  loadError?: Error;
+  exploreComponents: ODHApp[];
+  selectedComponent?: ODHApp;
+  updateSelection: (selectedId?: string | null) => void;
+};
+
+const ExploreApplicationsInner: React.FC<ExploreApplicationsInnerProps> = React.memo(
```

Comment:
The watcher memoizes the applications. However, since we sort the applications the watcher would think the applications changed if we didn't create the memoized sorted list and we would redraw all apps every time the fetch occurred even though nothing changed.

I've Memo'd the inner component because it is redrawing on every fetch as well though nothing has changed. I haven't quite figured out why, but I have been unable to prevent it without the `memo`.

Could do separate files, but really the intent is that this is one component. Unfortunately I've had to break it up in order to prevent the unnecessary redraws.